### ISSUE TITLE
Improve Cassie OSC walking controller

### ISSUE THREAD

**Issue by yminchen** [DAIRLab/dairlib#120]
The walking controller ( #118) is sensitive to the swing foot height at the moment when the finite state machine switches to a new state. We can probably add a double support state to make it more robust.

We should probably also use Raibert style foot placement rather than adding deviation to the Capture Point.

**Comment by yminchen** [comment]
Added double support phase and fixed a bug in trajectory acceleration in #126. The walking looks much better.

**Comment by yminchen** [comment]
Added a neutral point stepping location in #193

**Comment by yminchen** [comment]
OSC swing foot tracking doesn't look good currently. Need to investigate.

**Comment by yminchen** [comment]
> OSC swing foot tracking doesn't look good currently. Need to investigate.

Fixed a bug in swing foot trajectory generation in branch `cassie_experimental`. We were generating the spline starting from stance foot accidentally. Bug location: https://github.com/DAIRLab/dairlib/blob/be53e61/systems/controllers/cp_traj_gen.cc#L107

**Comment by yminchen** [comment]
Benchmark of OSC solve time with OSQP warm start:
https://github.com/DAIRLab/dairlib/pull/186#issuecomment-667592013

**Comment by yminchen** [comment]
A few more changes that lead to successful stable walking last year: #202, #212, #224, #247, #258, #267, #270.
One issue faced at the end was the wide walking foot spread -- I found that the swing foot prediction based on LIPM wasn't great in the y direction because the centroidal angular momentum doesn't stay constant throughout a stance phase. To account for this in a quick way, I changed the parameter of LIPM to make the step width smaller.
 
Last Friday, Brian tried a new swing foot planner based on ALIP (A stands for angular momentum; developed by Yukai Gong). Without any tuning, the new foot step planner seemed to work great.
